tests/runtime/servicers/test_training_management_servicer.py

```diff
@@ -75,6 +79,46 @@ def test_training_runs(training_management_servicer, training_pool):
     assert response.state == TrainingStatus.COMPLETED.value
 
 
+def test_training_timestamps(training_management_servicer, training_pool):
+    """Check that the submission and completion timestamps are set correctly from the model
+    future."""
+    # Create a future and set it in the training manager
+    event = threading.Event()
+    model_future = MODEL_MANAGER.train(
+        SampleModule,
+        DataStream.from_iterable([]),
+        wait_event=event,
+    )
+
+    # send a request, check that the submission timestamp was set
+    request = TrainingInfoRequest(training_id=model_future.id).to_proto()
+    response = training_management_servicer.GetTrainingStatus(request, context=None)
+    response_dm: TrainingStatusResponse = TrainingStatusResponse.from_proto(response)
+    assert response_dm.submission_timestamp is not None
+    assert (
```

Comment:
what is the purpose of making sure that the time of this code and the time the training was submitted is less than 50 ms?

Reply:
If I had been going with a mock trainer I would set the timestamp explicitly and check for an exact result. But since I don't know what time exactly the local trainer started, I'm just checking that it was in the very recent past.

50ms could be pretty tight on some setups, but at least on my machine most tests take on the order of 1ms